.changeset/rude-laws-battle.md

```diff
@@ -0,0 +1,5 @@
+---
+'@xstate/react': major
+---
+
+The signatures of `useMachine` and `useService` integrating with `@xstate/fsm` were changed. They now only accept a single generic each (`TMachine` and `TService` respectively). This has been done to match their signatures with the related hooks that integrate with `xstate` itself.
```

Comment:
Can we remove `useService` in here, in favor of `useActor`? Or is there a reason we still need `useService`?

Reply:
There is no `useActor` for `@xstate/fsm` and, so far, we've been using "service" as the name for the thing returned from `interpret`. So this makes it specific for how it can actually be used today.

We can rename it though but at the type-level this only accepts `TService` now. At the runtime level, the internal `getSnapshot` is also tied to a service type. I think it might be better to leave it as-is for now and reconsider later.